sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -2554,7 +2554,27 @@ object functions {
    * @since 1.5.0
    */
   def split(str: Column, pattern: String): Column = withExpr {
-    StringSplit(str.expr, lit(pattern).expr)
+    StringSplit(str.expr, lit(pattern).expr, lit(-1).expr)
+  }
+
+  /**
+   * Splits str around pattern (pattern is a regular expression) up to `limit-1` times.
```

Comment:
Drop `up to `limit-1` times` in the first line? That's because the behaviour depends on values described below.